src/server/index.tsx

```diff
@@ -294,28 +291,56 @@ export const createServer = ({
       return res.sendFile(path.join(rootPath, 'public', 'favicon.ico'));
     });
 
+    // eslint-disable-next-line global-require, @typescript-eslint/no-var-requires
+    const zlib = require('zlib');
+
     app.use(
       createProxyMiddleware('/', {
         target: `http://localhost:${env.REACT_APP_CRA_PORT}`,
         // We need WebSocket for Hot Module Reload (HMR).
         ws: true,
+        // This means we handle the response in `onProxyRes()`.
+        selfHandleResponse: true,
         onProxyRes: (
           proxyResponse: http.IncomingMessage,
           req: http.IncomingMessage & RequestWithCookies,
           res: http.ServerResponse,
         ) => {
-          const contentType = proxyResponse.headers['content-type'] || '';
-
-          if (contentType.includes('text/html')) {
-            // eslint-disable-next-line no-param-reassign
-            proxyResponse.headers['cache-control'] = 'private, no-store';
-
-            modifyResponse(
-              res,
-              proxyResponse.headers['content-encoding'],
-              (body: string) => _injectAuthenticationToken(req, body, env),
-            );
-          }
+          const bodyChunks: Buffer[] = [];
```

Comment:
I rewrote that part without `modifyResponse` which was not working anymore on Node 14. This library hasn't been updated for a while and it's a kinda obscure one so I preferred to add the rewrite code here.

Reply:
Note: that rewrite part is needed because tests fail on node 14, which wasn't covered before but I do reproduce locally.

Reply:
yes